pkg/api/handlers/mcp.go

```diff
@@ -504,27 +554,54 @@ func mergeMCPServerManifests(existing, override types.MCPServerManifest) types.M
 }
 
 func (m *MCPHandler) CreateServer(req api.Context) error {
+	catalogID := req.PathValue("catalog_id")
+
 	var input types.MCPServer
 	if err := req.Read(&input); err != nil {
 		return err
 	}
 
-	t, err := getThreadForScope(req)
-	if err != nil {
-		return err
-	}
+	var server v1.MCPServer
 
-	server := v1.MCPServer{
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: system.MCPServerPrefix,
-			Namespace:    req.Namespace(),
-		},
-		Spec: v1.MCPServerSpec{
-			Manifest:                  input.MCPServerManifest,
-			MCPServerCatalogEntryName: input.CatalogEntryID,
-			ThreadName:                t.Name,
-			UserID:                    req.User.GetUID(),
-		},
+	if catalogID != "" {
+		var catalog v1.MCPCatalog
+		if err := req.Get(&catalog, catalogID); err != nil {
+			return err
+		}
+
+		if catalog.Spec.IsReadOnly {
+			return types.NewErrForbidden("cannot create MCP server in read-only catalog")
+		}
+
+		server = v1.MCPServer{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: system.MCPServerPrefix,
+				Namespace:    req.Namespace(),
+			},
+			Spec: v1.MCPServerSpec{
+				Manifest:                   input.MCPServerManifest,
+				SharedWithinMCPCatalogName: catalogID,
+				UserID:                     req.User.GetUID(),
+			},
```

Comment:
I set this up so that the admin must provide the full manifest, rather than referencing a catalog entry. Let me know if this needs to change.

Reply:
You're correct. I'll switch it over.

Reply:
Actually I just heard from Craig that servers within catalogs should always be created from scratch, and never from an existing catalog entry, so I will keep this as it currently is.